src/lib.rs

```diff
@@ -332,7 +332,35 @@ pub trait Rand : Sized {
 /// An automatically-implemented extension trait on [`RngCore`] providing high-level
 /// generic methods for sampling values and other convenience methods.
 /// 
-/// This is the primary trait to use when generating random values. Example:
+/// This is the primary trait to use when generating random values.
+/// 
+/// # Generic usage
+/// 
+/// The basic pattern is `R: Rng` and `rng: &mut R`. Some
+/// things are worth noting here:
+/// 
+/// - Since `Rng: RngCore` and every `RngCore` implements `Rng`, it makes no
+///   difference whether we use `R: Rng` or `R: RngCore`
+/// - `fn foo<R: Rng>(rng: &mut R)` will not match type-erased `Rng`s; in order
+///   to support dynamic dispatch use either
+///   `fn foo<R: Rng + ?Sized>(rng: &mut R)` or `fn foo<R: Rng>(rng: R)`
+///   (but see next point about the latter).
```

Comment:
> Added a bit more doc. I hope it's not too scary :scream:

You give the technical reasons, and that is supposed to be a plus for programmers. But maybe it is good to help users a little. Maybe add something like:

> Tip if you want to use `Rng` in trait bounds:
> The basic pattern is `fn foo<R: Rng + ?Sized>(rng: &mut R)`.
>
> This way implementations can use the methods from both `RngCore` and `Rng`, support dynamic dispatch (thanks to `?Sized`), and ...
>
> Technical details:
> ...